packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -0,0 +1,274 @@
+          }
+          const file = resp
+            .body
+            .recording_files
+            .find(file => fileId === file.id || fileId === file.file_type)
+          if (!file || !file.download_url) {
+            return this._downloadError(resp, done)
+          }
+          resolve(file.download_url)
+        })
+    })
+    downloadUrlPromise.then((downloadUrl) => {
+      this.client
+        .get(`${downloadUrl}?access_token=${token}`)
+        .request()
+        .on('response', (resp) => {
+          if (resp.statusCode !== 200) {
+            done(this._error(null, resp))
+          } else {
+            resp.on('data', (chunk) => done(null, chunk))
+          }
+        })
+        .on('end', () => {
+          done(null, null)
+        })
+        .on('error', (err) => {
+          logger.error(err, 'provider.zoom.download.error')
+          done(err)
+        })
+    })
+  }
+
+  size ({ id, token, query }, done) {
+    const meetingId = id
+    const fileId = query.recordingId
+    const GET_MEETING_FILES = `/meetings/${meetingId}/recordings`
+
+    return this.client
+      .get(`${BASE_URL}${GET_MEETING_FILES}`)
+      .auth(token)
+      .request((err, resp, body) => {
+        if (err || resp.statusCode !== 200) {
+          return this._downloadError(resp, done)
+        }
+        const file = resp
+          .body
+          .recording_files
+          .find(file => file.id === fileId || file.file_type === fileId)
+
+        if (!file) {
+          return this._downloadError(resp, done)
+        }
+        // timeline files don't have file size, but are typically small json files, should be much less than 1MB
+        const maxExportFileSize = 1024 * 1024
+        done(null, file.file_size || maxExportFileSize)
+      })
+  }
+
+  _initializeData (body, initialEnd = null) {
+    let end = initialEnd || moment()
+    let start = end.clone().date(1)
+
+    const accountCreation = adapter.getAccountCreationDate(body)
+    const defaultLimit = end.clone().subtract(DEFAULT_RANGE_MOS, 'months')
+    const limit = accountCreation > defaultLimit ? accountCreation : defaultLimit
+
+    const data = {
+      items: [],
+      username: adapter.getUserEmail(body)
+    }
+
+    while (start > limit) {
+      start = end.clone().date(1)
+      data.items.push({
+        isFolder: true,
+        icon: 'folder',
+        name: adapter.getDateName(start, end),
+        mimeType: null,
+        id: adapter.getDateFolderId(start, end),
+        thumbnail: null,
+        requestPath: adapter.getDateFolderRequestPath(start, end),
+        modifiedDate: adapter.getDateFolderModified(end),
+        size: null
+      })
+      end = start.clone().subtract(1, 'days')
+    }
+    data.nextPagePath = adapter.getDateNextPagePath(start)
+    return data
+  }
+
+  _adaptData (results) {
+    if (!results) {
+      return { items: [] }
+    }
+
+    const data = {
+      nextPagePath: adapter.getNextPagePath(results),
+      items: []
+    }
```

Comment:
I set username to null since the UI never seems to update this field (it looks like the providerviews only sets this once). Let me know if you'd like this to be the user's email address though and I can add an API call to retrieve that information for this field.

Reply:
makes sense, this is updated